src/views/catalog/CreateFromInstanceTypes/components/SelectInstanceTypeSection/components/CategoryCard/components/InstanceTypesMenu/CustomMenu.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { ElementType, useRef, useState } from 'react';
+
+import CustomMenuRenderer from '@catalog/CreateFromInstanceTypes/components/SelectInstanceTypeSection/components/CategoryCard/components/InstanceTypesMenu/CustomMenuRenderer';
+import EnhancedMenuToggle from '@catalog/CreateFromInstanceTypes/components/SelectInstanceTypeSection/components/CategoryCard/components/InstanceTypesMenu/EnhancedMenuToggle';
+import InstanceTypesMenu from '@catalog/CreateFromInstanceTypes/components/SelectInstanceTypeSection/components/CategoryCard/components/InstanceTypesMenu/InstanceTypesMenu';
+
+type CustomMenuProps = {
+  toggleComponent: ElementType;
+  onSelect: (...args: any[]) => void;
```

Comment:
do we must use any ?

Reply:
I left some of these components more general in case we need to use them in the future. The `CustomMenu` component doesn't need to know what the caller and the distant child menu component will need, so I think `any` is the best option here.